integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisManagementServiceIntegrationTest.java

```diff
@@ -76,184 +65,86 @@
-                  }
-                  res.readEntity(CatalogRoles.class).getRoles().stream()
-                      .filter(cr -> !cr.getName().equals("catalog_admin"))
-                      .forEach(
-                          cr ->
-                              newRequest(
-                                      "http://localhost:%d/api/management/v1/catalogs/"
-                                          + catalog.getName()
-                                          + "/catalog-roles/"
-                                          + cr.getName())
-                                  .delete()
-                                  .close());
-                }
-
-                Response deleteResponse =
-                    newRequest(
-                            "http://localhost:%d/api/management/v1/catalogs/" + catalog.getName())
-                        .delete();
-                if (deleteResponse.getStatus() != Response.Status.NO_CONTENT.getStatusCode()) {
-                  LoggerFactory.getLogger(getClass())
-                      .warn(
-                          "Unable to delete catalog {}: {}",
-                          catalog.getName(),
-                          deleteResponse.readEntity(String.class));
-                }
-                deleteResponse.close();
-              });
-    }
-    try (Response response = newRequest("http://localhost:%d/api/management/v1/principals").get()) {
-      response.readEntity(Principals.class).getPrincipals().stream()
-          .filter(
-              principal ->
-                  !principal.getName().equals(PolarisEntityConstants.getRootPrincipalName()))
-          .forEach(
-              principal -> {
-                newRequest(
-                        "http://localhost:%d/api/management/v1/principals/" + principal.getName())
-                    .delete()
-                    .close();
-              });
-    }
-    try (Response response =
-        newRequest("http://localhost:%d/api/management/v1/principal-roles").get()) {
-      response.readEntity(PrincipalRoles.class).getRoles().stream()
-          .filter(
-              principalRole ->
-                  !principalRole
-                      .getName()
-                      .equals(PolarisEntityConstants.getNameOfPrincipalServiceAdminRole()))
-          .forEach(
-              principalRole -> {
-                newRequest(
-                        "http://localhost:%d/api/management/v1/principal-roles/"
-                            + principalRole.getName())
-                    .delete()
-                    .close();
-              });
-    }
+    managementApi
+        .listCatalogs()
+        .forEach(
+            catalog -> {
+              // clean up the catalog before we try to drop it
+              // delete all the namespaces
+              catalogApi
+                  .listNamespaces(catalog.getName())
+                  .forEach(namespace -> catalogApi.deleteNamespaces(catalog.getName(), namespace));
```

Comment:
All Quarkus tests are failing here:

```
Handling runtimeException Namespace top_level is not empty
...
Expected :204
Actual   :400
<Click to see difference>

org.opentest4j.AssertionFailedError: 
expected: 204
 but was: 400
	at org.apache.polaris.service.it.env.CatalogApi.deleteNamespaces(CatalogApi.java:89)
```

I guess it is dependent on the order that catalogs are returned from `listCatalogs`?

Reply:
`top_level` is not from this test :thinking: I'll change the code to clean up only what was defined.

Reply:
Somehow it's working now 🤷‍♂️